### ISSUE TITLE
1e-15 cutoff values for Scale Divisors questionable

### ISSUE THREAD

**Issue by StephanSchaller** [Open-Systems-Pharmacology/MoBi#272]
Often, I get 1e-15 Scale Divisor (SDs) values for compounds that actually are zero which then lead to numerical issues for these compounds. These Problems disappear once I set these SDs to 1.

Maybe we should reconsider the logic for setting cutoff values to1e-15 vs. setting these SDs to 1.

**Comment by StephanSchaller** [comment]
> What do you mean by that?

I had a protein loaded via expression database with 0% expression in a compartment that then calculated a SD of 1e-15 for this protein within this compartment.

I'll upload a project with a more detailed analysis soon...

**Comment by msevestre** [comment]
Do you mean then amount/concentrate is zero in the specific container for a given molecule?

**Comment by StephanSchaller** [comment]
Yes, sorry for the confusion: amount/concentration is zero (or maybe not "zero enough" ...as said, will look into it in more detail)

**Comment by StephanSchaller** [comment]
[Ex_2_Adding_a_Tumor_7.zip](https://github.com/Open-Systems-Pharmacology/MoBi/files/2008263/Ex_2_Adding_a_Tumor_7.zip)

For the attached Project:
For the existing simulation, I have set Sacel Divisors, but adapted them manually (set all 1e-15 entries --> 1): the simulations look smooth.

1. If you now reset the SDs, you will see some numerical noise in one of the lower concentration time curves.
2. If you now recalculate the SDs the simulation will crash with a negative value Error for Stomach|Interstitial|TMEFF.

TMEFF is a Target Protein expresses in some Tissues but with 0% in Stomach. However, there is a synthesis and degradation reaction defined in all tissues, but it is dependent on the expression levels, so for stomach, both rates (Synthesis & degradation) should be zero
![image](https://user-images.githubusercontent.com/25477643/40107385-60f87456-58f8-11e8-98cd-a9f913568277.png)

Now my theory on why the 1e-15 SDs cause Problems:

- If you add the AMOUNT of Stomach|Interstitial|TMEFF to you simulated outputs (with reset SDs (all = 1)), it shows only noise at around 1e-27 ... this is about the "normal" numerical noise of the solver.
- If you now consider, that the volume of the Stomach|Interstitial|Volume is around 1e-5 l, it means, that CONCENTRATION (i.e. Amount/Volume) is around 1e-21.
- with a solver accuracy (reative) of 1e-5, a SD of 1e-15 puts this noise in the range of solver accuracy and the solver crashes....

Questions that emerge:
1. similar to questions in #194 : what values (amount/concentration) are the SDs calculated from and applied to?!
2. What is the "algorithm" for setting values to 1e-15. I understand, that entities with amount/concentration = 0 are set to 1, right? and entities with 1e-15 > amount/concentration > 0 are set to 1e-15?
3. maybe a better solution is to set SDs to 1 if amount/concentration < 1e-15 and set the rest to their calculated value 

What do you think?
